bots/notify/src/main/java/org/openjdk/skara/bots/notify/IssueUpdater.java

```diff
@@ -252,25 +255,40 @@ public void handleCommits(HostedRepository repository, Repository localRepositor
                     continue;
                 }
 
-                // The actual issue to be updated can change depending on the fix version
                 String requestedVersion = null;
-                if (setFixVersion) {
-                    requestedVersion = fixVersions != null ? fixVersions.getOrDefault(branch.name(), null) : null;
-                    if (requestedVersion == null) {
-                        try {
-                            var conf = localRepository.lines(Path.of(".jcheck/conf"), commit.hash());
-                            if (conf.isPresent()) {
-                                var parsed = JCheckConfiguration.parse(conf.get());
-                                var version = parsed.general().version();
-                                requestedVersion = version.orElse(null);
+                if (prOnly) {
+                    var pullRequests = issue.links().stream()
+                                            .filter(link -> link.title().orElse("notitle").equals("Review"))
+                                            .filter(link -> link.summary().orElse("nosummary").startsWith(repository.name() + "/"))
+                                            .map(link -> link.summary().orElseThrow().substring(repository.name().length() + 1))
+                                            .map(repository::pullRequest)
+                                            .filter(pr -> pr.targetRef().equals(branch.name()))
+                                            .collect(Collectors.toList());
```

Comment:
Use [Stream.count](https://docs.oracle.com/en/java/javase/13/docs/api/java.base/java/util/stream/Stream.html#count()) here instead of materializing the list? You would have to change the following `if` to `if (count == 0L) {`.

Reply:
Good point, will fix!